### DISCUSSION TITLE
problems in vitepress

### DISCUSSION THREAD

**Discussion by mancuoj** [giscus/giscus#830] [Q&A] [answered]
First of all, thank you very much for your open source project!

I am new in frontend, and I want to use `key` to reload the giscus component when user toggles dark mode, but when I toggle the mode, the display of this component is no longer complete 

![image](https://user-images.githubusercontent.com/45707684/208154629-5ee6173a-4ed6-41e5-b1b1-256bea20f63b.png)

this is [my code](https://github.com/mancuoj/csdiy/blob/main/docs/.vitepress/theme/components/Comment.vue), and preview doc link: https://cs.hrh.icu/cs/intro

```vue
<script setup>
import Giscus from '@giscus/vue'
import DefaultTheme from 'vitepress/theme'
import { useData } from 'vitepress'

const { isDark } = useData()
const { Layout } = DefaultTheme
</script>

<template>
  <Layout>
    <template #doc-after>
      <div id="giscus" :key="isDark">
        <Giscus ... lang="zh-CN" loading="lazy" :theme="[isDark ? 'transparent_dark' : 'light_tritanopia']" />
      </div>
    </template>
  </Layout>
</template>

<style scoped>
#giscus {
  margin-top: 3.2rem;
}
</style>
```

**Comment by laymonage** [comment]
You shouldn't need to use `key`, the component will automatically change the theme when you change the `theme` prop...

**Reply by mancuoj** [reply]
Thank you for your answer, but I didn't use `key` to re-render the component at the beginning, and I started to use it after I found that the theme could not be changed automatically.

![image](https://user-images.githubusercontent.com/45707684/208301563-01ce5ce1-0473-48c7-8ed5-e27b72073473.png)

and you can preview in my latest update.

**Comment by laymonage** [comment, marked as answer]
You need to use `computed` for your use case, because the `theme` is computed from `isDark`. Without `computed`, the value is only calculated on initial render.

Fix: https://github.com/mancuoj/csdiy/pull/20

**Reply by mancuoj** [reply]
Thanks so much!
